src/dialect/snowflake.rs

```diff
@@ -93,6 +102,170 @@ impl Dialect for SnowflakeDialect {
     }
 }
 
+pub fn parse_create_table(
+    or_replace: bool,
+    local: bool,
+    temporary: bool,
+    _volatile: bool,
+    transient: bool,
+    parser: &mut Parser,
+) -> Result<Statement, ParserError> {
+    let if_not_exists = parser.parse_keywords(&[Keyword::IF, Keyword::NOT, Keyword::EXISTS]);
+    let table_name = parser.parse_object_name(false)?;
+
+    let mut builder = CreateTableBuilder::new(table_name)
+        .or_replace(or_replace)
+        .if_not_exists(if_not_exists)
+        .temporary(temporary)
+        .transient(transient)
+        .hive_formats(Some(Default::default()));
```

Comment:
The `hive_formats` is tested in a generic test against all dialects, since I don't know which dialect uses the `hive_format` I added the default value in the return struct to keep that test happy.